docs/proposals/20220718-workload-update-model.md

```diff
@@ -0,0 +1,124 @@
+
+### Goals
+
+- Provide OTA update: users can control the daemonset application updates through the API exposed on the edge nodes.
+- Add a new controller: for `ready` nodes, execute the daemosnet application updates directly; for `not-ready` nodes, wait for nodes' status to turn ready before executing the update process.
+
+## Proposal
+
+### User Stories
+
+#### Story 1
+
+- In some node-independent scenarios, such as a user owning a node, for example, on a Tesla car, the owner wants to independently update his application, instead of being controlled by the cluster administrator. At this time, the user can complete the application update process through a two-step process.
+  1. Check whether the latest version of the application is available.
+  2. Confirm the execution of application update.
+
+#### Story 2
+
+- In a openyurt cluster, there are some edge nodes, and some node's status are `not-ready` due to the network, when the administrator updates the application version, the administrator expects that for the `ready` edge nodes, the application can be updated to the latest version smoothly; for the `not-ready` edge nodes, they do not block the update process of other nodes.
+
+### Implementation Details
+
+Note: No modification to Kubernetes, keeping it non-invasive, no modification to the native workload controllers.
+
+#### 1 OTA-Update
+
+<div align="center">
+  <img src="../img/OTA.png" width=50% title="OTA">
+</div>
+
+**Pre-requisites**: Set application's updateStrategy to `type: OnDelete`; add a new annotation(`apps.openyurt.io/upgrade-strategy: ota`) for DaemonSet
+
+1. Users query whether a newer version of the application is available via OTA module's API.
+2. OTA module accesses Kubernetes APIServer through client-go library and get the current application's version.
+3. If update is available, users can choose to apply update. (This is only from the user's perspective)
+4. OTA module removes the selected pod from the current node.（This step is achieved by calling the Kubernetes interface）'
+5. Kubelet on this node will synchronize pods information from Kubernetes APIServer.
+6. Kubelet is responsible for creating the latest version of pod.
+
+<div align="center">
+  <img src="../img/process.png" width=40% title="OTA Process">
+</div>
+
+##### OTA API
+
+1. get all pods on edge node
+
+```text
+GET /pods HTTP/1.1
+```
+
+1. upgrade specified pod on edge node
+
+```text
+POST /openyurt.io/v1/namespaces/{ns}/pods/{podname}/upgrade  HTTP/1.1
+````
+
+#### 2 New-controller
+
+<div align="center">
+  <img src="../img/xxx-controller.png" width=40% title="new-controller">
+</div>
+
+##### Resources of interest
+
+- Daemonset
+- Node
+
+##### Steps
+
+**Pre-requisites**: Set application's updateStrategy to `type: OnDelete`; add a new annotation(`apps.openyurt.io/upgrade-strategy: auto`) for DaemonSet
```

Comment:
What will be the impact if `type: OnDelete` is not set? Do you need Webhook verification？

Reply:
@kadisi I agree with you that it's better to add validating webhook for upgrade strategy setting, but we can add webhook when coding, so i will merge this pull request at first.